spirv_hlsl.cpp

```diff
@@ -2483,6 +2497,30 @@ void CompilerHLSL::analyze_meshlet_writes(uint32_t func_id, uint32_t id_per_vert
 					bool already_declared = false;
 					auto builtin_type = BuiltIn(get_decoration(var->self, DecorationBuiltIn));
 
+					if (op == OpAccessChain && i.length > 4)
+					{
+						auto &type = get_variable_element_type(*var);
+						auto *c = maybe_get<SPIRConstant>(ops[4]);
+						if (c != nullptr && c->scalar() == 0 &&
+						    has_member_decoration(type.self, 0, DecorationBuiltIn) &&
```

Comment:
This does not consider that you can have plain Position variables.

This check is also too naive. A more correct solution involves:

- An access chain that is created from another access chain will miss this check. It should inherit the access_to_meshlet_position in this case. Test case: Access chain into Position, then access chain into scalar.
- A more correct solution should traverse the access chain and insert the unordered_set when it comes across anything related to positions.
- OpCopyObject needs to inherit the flag.
- Missing OpInBoundsAccessChain